### ISSUE TITLE
"Not a valid format" when parsing an ISO 8601 string

### ISSUE THREAD

**Issue by tspoke** [korlibs/klock#123]
Hi,

I'm getting a parsing issue from a raw string to a DateTime, hereafter my code and results.
**getBirthdayDate()** is implemented in KMP but it is called from an android app.

The KMP source code is very simple : 
```
// birthday == "1989-01-01T10:00:00.000Z"
fun getBirthdayDate(): DateTimeTz? {
        birthDay?.let {
            return DateTime.parse(it)
        }
        return null
}
```

On android I get this error :
```
com.soywiz.klock.DateException: Not a valid format: '1989-01-01T10:00:00.000Z' for 'yyyy-MM-dd'T'HH:mm:ssXXX'
        at com.soywiz.klock.DateFormatKt.parse(DateFormat.kt:32)
        at com.soywiz.klock.DateFormat$Companion.parse(DateFormat.kt:19)
        at com.soywiz.klock.DateTime$Companion.fromString(DateTime.kt:182)
```

I tried to remove the timezone Zulu character but I still have the same error :
```
com.soywiz.klock.DateException: Not a valid format: '1989-01-01T10:00:00.000' for 'yyyy-MM-dd'T'HH:mm:ssXXX'
        at com.soywiz.klock.DateFormatKt.parse(DateFormat.kt:32)
        at com.soywiz.klock.DateFormat$Companion.parse(DateFormat.kt:19)
        at com.soywiz.klock.DateTime$Companion.fromString(DateTime.kt:182)
```

**If I remove the milliseconds, it works has expected. Is it by design ?**

Version : 1.11.12

**Comment by RezMike** [comment]
The [default formats](https://github.com/korlibs/klock/blob/master/klock/src/commonMain/kotlin/com/soywiz/klock/DateFormat.kt#L9) don't take milliseconds into account. Try this line of code:
```
DateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSz").parse(it)
```

**Comment by tspoke** [comment]
I was expecting from a date-time library to be able to parse correctly at least an ISO 8601 format (with optional ms). Is it by design ? 
IMO this format should be handled correctly by default.

**Comment by RezMike** [comment]
Good point. I've fixed it here #125
